pkg/ctl/cmdutils/iamserviceaccount_filter.go

```diff
@@ -0,0 +1,157 @@
+	return &IAMServiceAccountFilter{
+		Filter: &Filter{
+			ExcludeAll:   false,
+			includeNames: sets.NewString(),
+			excludeNames: sets.NewString(),
+		},
+	}
+}
+
+// AppendGlobs appends globs for inclusion and exclusion rules
+func (f *IAMServiceAccountFilter) AppendGlobs(includeGlobExprs, excludeGlobExprs []string, serviceAccounts []*api.ClusterIAMServiceAccount) error {
+	if err := f.AppendIncludeGlobs(serviceAccounts, includeGlobExprs...); err != nil {
+		return err
+	}
+	return f.AppendExcludeGlobs(excludeGlobExprs...)
+}
+
+// AppendIncludeGlobs sets globs for inclusion rules
+func (f *IAMServiceAccountFilter) AppendIncludeGlobs(serviceAccounts []*api.ClusterIAMServiceAccount, globExprs ...string) error {
+	return f.doAppendIncludeGlobs(f.collectNames(serviceAccounts), "iamserviceaccount", globExprs...)
+}
+
+// SetExcludeExistingFilter uses stackManager to list existing nodegroup stacks and configures
+// the filter accordingly
+func (f *IAMServiceAccountFilter) SetExcludeExistingFilter(stackManager *manager.StackCollection, clientSet kubernetes.Interface, serviceAccounts []*api.ClusterIAMServiceAccount, overrideExistingServiceAccounts bool) error {
+	if f.ExcludeAll {
+		return nil
+	}
+
+	existing, err := stackManager.ListIAMServiceAccountStacks()
+	if err != nil {
+		return err
+	}
+
+	if !overrideExistingServiceAccounts {
+		err := f.ForEach(serviceAccounts, func(_ int, sa *api.ClusterIAMServiceAccount) error {
+			exists, err := kubernetes.CheckServiceAccountExists(clientSet, sa.ObjectMeta)
+			if err != nil {
+				return err
+			}
+			if exists {
+				existing = append(existing, sa.NameString())
+			}
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return f.doSetExcludeExistingFilter(existing, "iamserviceaccount")
+}
+
+// SetIncludeOrExcludeMissingFilter uses stackManager to list existing iamserviceaccount stacks and configures
+// the filter to either explictily exluce or include iamserviceaccounts that are missing from given serviceAccounts
+func (f *IAMServiceAccountFilter) SetIncludeOrExcludeMissingFilter(stackManager *manager.StackCollection, includeOnlyMissing bool, serviceAccounts *[]*api.ClusterIAMServiceAccount) error {
+	existing, err := stackManager.ListIAMServiceAccountStacks()
+	if err != nil {
+		return err
+	}
+
+	remote := sets.NewString(existing...)
+	local := sets.NewString()
+
+	for _, localServiceAccount := range *serviceAccounts {
+		localServiceAccountName := localServiceAccount.NameString()
+		local.Insert(localServiceAccountName)
+		if !remote.Has(localServiceAccountName) {
+			logger.Info("iamserviceaccounts %q present in the given config, but missing in the cluster", localServiceAccountName)
+			f.AppendExcludeNames(localServiceAccountName)
+		} else if includeOnlyMissing {
+			f.AppendExcludeNames(localServiceAccountName)
+		}
+	}
+
+	for remoteServiceAccountName := range remote {
+		if !local.Has(remoteServiceAccountName) {
```

Comment:
You can reduce nesting and therefore improve readability by using the negated condition:

```
for ..... {
  if local.Has(remoteServiceAccountName) {
    continue
  }
  if !includeOnlyMissing {
    continue
  }
  meta, err := api.ClusterIAMServiceAcccountNameStringTooObjectMeta(......)
  .....
}
```